learners/BF_measles.Rmd

```diff
@@ -0,0 +1,884 @@
+          beta,
+          ") for different age groups."
+        )
+      )
+  ) +
+  # Adjust legend display to fit in two rows
+  guides(
+    color = guide_legend(nrow = 2)
+  )
+```
+
+## Vaccination Model
+
+An effective vaccine for measles exists, known as the measles, mumps,
+and rubella (MMR) vaccine. The efficacy of the measles component of the
+MMR vaccine varies slightly depending on the age at which it is
+administered and the number of doses received. For example, the
+vaccine's efficacy is approximately $93\%$, $97\%$, $95\%$, $94\%$, and
+$85\%$ for the age groups $[0\to 15)$, $[15\to 25)$, $[25\to 55)$,
+$[55\to 65)$, and $65+$, respectively. The `{epidemics}` package enables
+the incorporation of vaccination interventions into modeling, as
+demonstrated in the code chunk below.
+
+
+```{r}
+vaccinate_elders <- epidemics::vaccination(
+  name = "Vaccination compagain",
+  time_begin = matrix(30, nrow(contact_matrix)),
+  time_end = matrix(90, nrow(contact_matrix)),
+  nu = matrix(c(0.93, 0.97, 0.95, 0.90, 0.95))
+)
+
+# Run the epidemic model
+output3 <- epidemics::model_default(
+  population = bf_population,
+  transmission_rate = 15 / 5,  # R0/infectious period = 15/5 = 3
+  infectiousness_rate = 1 / 8,  # 1/latent period (pre-infectious phase)
+  recovery_rate = 1 / 5,  # 1/infectious period
+  vaccination = vaccinate_elders,
+  time_end = 120,  # Simulation over 120 days
+  increment = 1  # Daily increments
+)
+
+# Plot the epidemic curve
+output3 %>%
+  filter(compartment %in% c("exposed", "infectious")) %>%
+  ggplot(aes(
+    x = time,
+    y = value,
+    col = demography_group,
+    linetype = compartment
+  )) +
+  geom_line(linewidth = 1.2) +
+  scale_y_continuous(labels = scales::comma) +
+  scale_colour_brewer(palette = "Dark2", name = "Age group") +
+  expand_limits(y = c(0, 3e3)) +
+  coord_cartesian(expand = FALSE) +
+  theme_bw() +
+  theme(
+    legend.position = "top",
+    legend.text = element_text(size = 10, hjust = 0, lineheight = 121)
+  ) +
+  labs(
+    x = "Simulation time (days)",
+    linetype = "Compartment",
+    y = "Individuals",
+    caption =
+      expression(
+        paste(
+          "SIER with vaccine  showing the trajectories of the number of ",
+          "exposed and infected individuals for different age groups."
+        )
+      )
+  ) +
+  guides(
+    color = guide_legend(nrow = 2)
+  )
+```
+
+We can visualize the outcomes of both the baseline model and the
+vaccine-intervention model to highlight the differences in infection
+dynamics. This comparison allows us to observe how vaccination impacts
+the spread of the disease, particularly in terms of reducing new
+infections, altering the epidemic curve, and potentially preventing
+outbreaks across different age groups. By contrasting these two models,
+we can better understand the effectiveness of vaccination strategies in
+controlling the epidemic.
+
+
+```{r}
```

Comment:
Something looks odd in these plots - the number of infections is far higher than the other scenarios considered?

Reply:
Thanks for this subtle error. The `compartments_from_susceptible` argument was `NUll`.